ssl/t1_lib.c

```diff
@@ -3452,3 +3450,27 @@ size_t ssl_hmac_size(const SSL_HMAC *ctx)
     return 0;
 }
 
+#ifndef OPENSSL_NO_EC
+int ssl_get_EC_curve_nid(const EVP_PKEY *pkey)
+{
+    int ret = NID_undef;
+
+    if (EVP_PKEY_id(pkey) != EVP_PKEY_KEYMGMT) {
+        if (EVP_PKEY_base_id(pkey) == EVP_PKEY_EC) {
+            EC_KEY *ec = EVP_PKEY_get0_EC_KEY(pkey);
```

Comment:
Noooooo!!!!!!!!!! All EC_KEY functions are about to be deprecated (#13139) we must not introduce new usage!

Reply:
I've updated this function to use existing `EVP_PKEY_CTX_get_group_name()`...  Muuuuch neater